pkg/ebpf/c/bpf_telemetry.h

```diff
@@ -5,69 +5,62 @@
 #include "telemetry_types.h"
 #include "map-defs.h"
 
+// this macro reads the stack slot at offset 512
+// If everything went okay, then it should have the pointer
+// to the telemetry map.
+#define FETCH_TELEMETRY_BLOB() ({                   \
+    instrumentation_blob_t *__tb;                   \
+    asm("%0 = *(u64 *)(r10 - 512)" : "=r"(__tb));   \
+    __tb;                                           \
+})
```

Comment:
Does this create two instances of `instrumentation_blob_t *` (in the macro, and outside)? I wonder if you took the target as an argument, if that would simplify things.
```suggestion
#define FETCH_TELEMETRY_BLOB(__tb) ({                   \
    asm("%0 = *(u64 *)(r10 - 512)" : "=r"(__tb));   \
})
```

Reply:
The compiler optimizes it to a single instruction. Also the syntax looks neater IMO if it looks like `FETCH_TELEMETRY_BLOB` is returning a value rather than "setting" it